### PR TITLE
feat: migrate countries from types

### PR DESCRIPTION
**Pull request by jamie-lynch**
## What does this change?

* Migrates the Country enum and CountryCode type from `@guardian/types` to `@guardian/libs`
* Renames the `Country` enum to `CountriesByCountryCode`

## Why?

See https://github.com/guardian/types/issues/136

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -6,6 +6,7 @@ export {
 	setCookie,
 	setSessionCookie,
 } from './cookies';
+export type { Country, CountryCode } from './countries';
```
Comment (jamie-lynch) [inline src/index.ts:9]: Should `Country` be exported as a type here?

Comment (sndrs) [reply src/index.ts]: after chatting offline, no :)

File: src/countries.README.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# `countries`
```
Comment (sndrs) [inline src/countries.README.md:1]: think this needs updating?

**Review by sndrs — CHANGES_REQUESTED** [review]

File: src/countries.README.md (hunk 2 above)
Comment (jamie-lynch) [reply src/countries.README.md:1]: I've just updated the usage example to `CountriesByCountryCode`. Do you think the filename should be changed too?

Comment (sndrs) [reply src/countries.README.md:1]: i've been umming and ahhing about this.

should we expand this to look something like:

```typescript
export const countries = {
	afghanistan: {
		name: 'Afghanistan',
		countryCode: 'AF',
	},
	bolivia: {
		name: 'Bolivia (Plurinational State of)',
		countryCode: 'BO',
	},
} as const;

export type Country = keyof typeof countries;
export type CountryCode = typeof countries[Country]['countryCode'];
export type CountryName = typeof countries[Country]['name'];

let getCountryByCodeCache: {
	[prop in CountryCode]?: CountryName;
} = {};

export const getCountryNameByCountryCode = (countryCode: CountryCode) =>
	(getCountryByCodeCache[countryCode] ||= Object.values(countries).find(
		(country) => country.countryCode === countryCode,
	)?.name);
```

so then

```typescript
import { countries, getCountryNameByCountryCode } from '@guardian/libs';

const afghanNameFromCode = getCountryNameByCountryCode('AF')
const afghanName = countries.afghanistan.name
```

Comment (sndrs) [reply src/countries.README.md:1]: (and keep the filename)

Comment (sndrs) [reply src/countries.README.md:1]: cc @mxdvl

Comment (jamie-lynch) [reply src/countries.README.md:1]: This makes sense to me. I'll have a go at implementing it and update this PR.

Comment (mxdvl) [reply src/countries.README.md:1]: Don’t have a strong feeling about this… I guess it’s worth mentioning we already have a mapping between country codes and country names in `frontend` here: https://github.com/guardian/frontend/blob/bb292f016a0089f809dffe2851436efe66cd15fe/static/src/javascripts/lib/geolocation.ts

**Review by sndrs — APPROVED** [review]

**Review by sndrs — APPROVED** [review]

**Pull request merged**